cfme/scripting/bz.py

```diff
@@ -0,0 +1,149 @@
+"""
+Scripts for dealing with Bugzilla metadata, listing BZs with coverage, and setting qe_test_coverage
+flag. This script looks at test-case metadata for the flags "automates" and "coverage", and fetches
+information about the BZs listed there.
+
+For usage see, "miq bz --help"
+
+The basic usage of this command is via
+```python,
+miq bz <command> <directory>
+```
+where <command> is one of report, list, or coverage, and <directory> is the testing directory, e.g.
+"cfme/tests/control/"
+
+To list BZs that this script would set coverage for, you can do,
+```python,
+miq bz coverage <directory>
+```
+This will print out the ids of BZs that could have qe_test_coverage switched to '+'
+To then set qe_test_coverage on those BZs, you can do,
+```python,
+miq bz coverage --set <directory>
+```
+"""
+import os
+from collections import namedtuple
+
+import click
+import pytest
+import yaml
+
+from cfme.utils.blockers import BZ
+from cfme.utils.log import logger
+
+
+def get_report(directory):
+    pytest.main([
+        "--use-provider", "complete",
+        "--long-running",
+        "--use-template-cache",
+        "--collect-only",
+        "--generate-bz-report", directory
+    ])
+    # read the generated yaml
+    with open("bz-report.yaml", "r") as stream:
+        info = yaml.load(stream, Loader=yaml.BaseLoader)
```

Comment:
I am using `load` here because this yaml file has objects in it like `!!python/object:xmlrpclib.DateTime` that `safe_load` doesn't recognize. That is also why I have `Loader=yaml.BaseLoader`, so that it parses those as strings. Do you know of a way I can work around this?

Reply:
aahhh we are using old pckg `PyYAML` need to move on `ruamel.yaml`.